[B5] Exports: list and download pages

orangejenny opened the pull request:
## Product Description

User-facing changes are minimal.

|Before|After|
|---|---|
|![Screenshot 2024-10-01 at 5 11 55 PM](https://github.com/user-attachments/assets/41535647-7a3e-43e7-93f1-686a63b1cd3f)|![Screenshot 2024-10-01 at 5 14 05 PM](https://github.com/user-attachments/assets/cec53afb-ad87-493d-8e4c-5e6f03a295bf)|
|![Screenshot 2024-10-01 at 5 12 32 PM](https://github.com/user-attachments/assets/520bea0b-48f7-4ff9-a416-e7ad0d0dbbb4)|![Screenshot 2024-10-01 at 5 13 57 PM](https://github.com/user-attachments/assets/382c0f18-a144-41ae-88ac-6cfffedbf7f0)|
|![Screenshot 2024-10-01 at 5 12 43 PM](https://github.com/user-attachments/assets/bda73d8d-dd62-4126-9403-783c346ccf54)|![Screenshot 2024-10-01 at 5 14 56 PM](https://github.com/user-attachments/assets/ee84bc31-810e-4b6b-829d-885ca59ac81e)|
|![Screenshot 2024-10-01 at 5 12 54 PM](https://github.com/user-attachments/assets/75cdb7ca-2139-462a-9b70-aad594d03be6)|![Screenshot 2024-10-01 at 5 14 32 PM](https://github.com/user-attachments/assets/f2c290c1-3159-417b-bc21-f72170334b38)|


## Safety Assurance

### Safety story
UI-level changes. Requesting QA.

### Automated test coverage
Not much. There are some exports js tests.

### QA Plan
https://dimagi.atlassian.net/browse/QA-7132 - joint QA with https://github.com/dimagi/commcare-hq/pull/35170

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

orangejenny committed:
Turned on B5 for view

orangejenny committed:
Updated modal headers

orangejenny committed:
Styling updates, mostly removing inline styles

orangejenny committed:
Removed bulk delete tooltip

This button pops up a confirmation modal, that's enough.

orangejenny committed:
Replaced bulk export tooltip with help bubble

This is the stanard pattern in HQ. Bonus, it doesn't need js initialization.

orangejenny committed:
Fixed layout of bulk buttons

orangejenny committed:
Updated dropdown placeholder to match form label

orangejenny committed:
Minor text updates

.text-warning isn't readable on a white background, be lazy and change it to .text-danger.
Shorten the 'Copy & Edit Feed' text so it doesn't spill onto two lines.
Remove .btn-link which is a weird affordance, since links don't typically open modals.

orangejenny committed:
Made basic CSS updates to checkboxes, etc.

orangejenny committed:
Removed TODO

This crispy form is just fine.

orangejenny committed:
Replaced .well with .card, removed TODO

orangejenny committed:
Fixed data-target attribute names

orangejenny committed:
Updated modals

orangejenny committed:
Turned on B5 for download view

orangejenny committed:
Removed unnecessary form CSS classes

orangejenny committed:
Migrated export_download_progress.html, all minor updates

orangejenny committed:
Migrated export_download_prepare.html

orangejenny committed:
Moved date range picker to B5

orangejenny committed:
Fixed lint errors

orangejenny committed:
"Bootstrap 5 Migration - Rebuilt diffs"

AmitPhulera commented on corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_variables.scss:
```diff
@@ -10,6 +10,8 @@ $container-max-widths: (
 );
 
+// Spacing override
+$enable-negative-margins: true;
```
I don't understand what impact will it have 😅

orangejenny replied:
Check [this](https://getbootstrap.com/docs/5.0/utilities/spacing/#negative-margin) out: B5's negative margin utility classes are disabled by default, though I don't know why.

So this was needed to use the `mt-n3` added in [this commit](https://github.com/dimagi/commcare-hq/commit/2760a7a0a52398b41a61ea32de9cae9fb6c94f54#diff-6dca3dbb115515abc14b50769fcf871ebe30b05e2f64645034a385a6552eb6ceL41).

AmitPhulera replied:
That makes sense. Thanks for explaining.

AmitPhulera commented on corehq/apps/export/templates/export/partials/bootstrap5/table.html:
```diff
@@ -257,7 +257,7 @@ <h4 class="modal-title">
                       data-bind="visible: emailedExport.canUpdateData,
                                  attr: {
-                                   'data-target': '#modalRefreshExportConfirm-' + id() + '-' + emailedExport.groupId()
+                                   'data-bs-target': '#modalRefreshExportConfirm-' + id() + '-' + emailedExport.groupId()
```
Does this change is flagged by the migration tool?

orangejenny replied:
It originally didn't get flagged because the migration tool was only looking for regular data attributes set directly in elements, like `<div data-target="#myModal">`, but https://github.com/dimagi/commcare-hq/pull/35160 updates the tool to address this.

AmitPhulera replied:
I see. The PR was added later 😅

AmitPhulera reviewed (approved):
Changes looked good and made sense based on screenshots provided in the description. Should be good to go once QA is done.

orangejenny committed:
Merge branch 'master' into jls/b5-export-list

orangejenny committed:
Bootstrap 5 Migration - Rebuilt diffs

orangejenny committed:
Fixed select2s in modal

https://select2.org/troubleshooting/common-problems

orangejenny committed:
Moved showLink call inside ko with block.

The showLink reference needs to be inside of the with block, because it's an attribute of the exportFeedUrl model.

orangejenny committed:
Merge branch 'master' into jls/b5-export-list

orangejenny committed:
"Bootstrap 5 Migration - Rebuilt diffs"

millerdev reviewed (approved):
Review of changes since Amit's approval.

orangejenny committed:
Fixed text alignment of bulk actions buttons.

Text alignment classes are not for layout. Oops.

orangejenny committed:
"Bootstrap 5 Migration - Rebuilt diffs"
